### PR TITLE
Added configuration of max limits for receiver signals.

### PR DESCRIPTION
**Pull request by superdupersvard**
New feature according to issue #102

### CODE REVIEW

File: fcb-source/fcb/src/flight_control.c [hunk 1]
```diff
@@ -239,47 +236,70 @@ static void SetReferenceSignals(void) {
...
+		RefSignals.YawAngleRate = -RefSignalsLimits.YawAngleRate*(rudder - RECEIVER_TO_REFERENCE_ZERO_PADDING)
 				/ (INT16_MAX - RECEIVER_TO_REFERENCE_ZERO_PADDING);
 	} else {
-		RefSignals.YawAngleRate = -MAX_YAW_RATE*(rudder + RECEIVER_TO_REFERENCE_ZERO_PADDING)
+		RefSignals.YawAngleRate = -RefSignalsLimits.YawAngleRate*(rudder + RECEIVER_TO_REFERENCE_ZERO_PADDING)
 				/ (-INT16_MIN - RECEIVER_TO_REFERENCE_ZERO_PADDING);
 	}
 }
 
+void setMaxLimitForReferenceSignal(float maxZVelocity, float maxRollAngle, float maxPitchAngle, float maxYawAngleRate) {
```
Comment (DanielStenis) [inline fcb-source/fcb/src/flight_control.c]: Perhaps use float32_t type for extra clarity, but no real difference.

File: fcb-source/communication/com_cli.c [hunk 2]
```diff
@@ -266,6 +268,21 @@ static const CLI_Command_Definition_t getRefSignalsCommand = { (const int8_t * c
         0 /* Number of parameters expected */
 };
 
+/* Structure that defines the "set-max-limit-receiver-signals" command line command. */
```
Comment (DanielStenis) [inline fcb-source/communication/com_cli.c]: Should be ReferenceSignals and set-max-limit-reference-signals etc.

File: fcb-source/utilities/src/flash.c [hunk 3]
```diff
@@ -69,6 +69,36 @@ FlashErrorStatus WriteCalibrationValuesToFlash( const Receiver_CalibrationValues
 	return status;
 }
 
+/*
+ * @brief  Reads previously stored receiver max limits from flash memory
+ * @param  receiverMaxLimits : Pointer to receiver max limits struct to which values will enter
+ * @retval FLASH_OK if values read succesfully from flash, else FLASH_ERROR
+ */
+FlashErrorStatus ReadReceiverMaxLimitsFromFlash(RefSignals_TypeDef* receiverMaxLimits) {
```
Comment (DanielStenis) [inline fcb-source/utilities/src/flash.c]: Reference signals, not receiver signals here as well

File: fcb-source/utilities/src/flash.c [hunk 4]
```diff
@@ -214,7 +244,7 @@ static FlashErrorStatus ReadFlashPage(uint8_t * readData, const uint8_t pageNbr)
  * @retval word value as an unsigned 32-bit integer
  */
 static uint32_t ReadFlashWord(const uint8_t pageNbr, const uint16_t wordNbr) {
```
Comment (DanielStenis) [inline fcb-source/utilities/src/flash.c]: Now the variable wordNbr will not actually be the word number, but rather the byte number?

**Comment by DanielStenis** [comment]
Everything seems to work as intended, nice work!

A few mix ups in the code where "receiver signals" is errenously used for the "reference signals" limits (they are different things, receiver signals is the values from the RC receiver, reference signals is the values fed in to the control algorithms). 

E.g. "set-max-limit-receiver-signals" should be "set-max-limit-reference-signals".

Perhaps we should also have a "reset-max-limit-reference-signals" command to reset to defefalt values?

File: fcb-source/communication/com_cli.c (hunk 2 above)
Comment (superdupersvard) [inline fcb-source/communication/com_cli.c]: Done

File: fcb-source/utilities/src/flash.c (hunk 3 above)
Comment (superdupersvard) [inline fcb-source/utilities/src/flash.c]: Done

File: fcb-source/utilities/src/flash.c (hunk 4 above)
Comment (superdupersvard) [inline fcb-source/utilities/src/flash.c]: Done

**Pull request merged**